besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -140,6 +142,9 @@ public abstract class BesuControllerBuilder implements MiningParameterOverrides
       Collections.emptyList();
   protected EvmConfiguration evmConfiguration;
   protected int maxPeers;
+  protected boolean isChainDataPruningEnabled;
+  protected long chainDataPruningBlocksRetained;
+  protected long chainDataPruningFrequency;
```

Comment:
Could maybe wrap in a ChainDataPruningConfiguration object?

Reply:
Done